### PR TITLE
Fixes #142 - Create integration test framework

### LINKED ISSUE #142: Add integration test capability to migrator
Getting this in dev right now, over and over:

```
2015-03-27 15:42:00,037 [WorkerMonitor] com.redhat.lightblue.migrator.consistency.ConsistencyCheckerDaemon INFO - Starting ConsistencyCheckerDaemon
2015-03-27 15:42:00,038 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - From CLI - consistencyCheckerName: hystrix hostName: lightbluehystrix1.dev.a1.vary.redhat.com
2015-03-27 15:42:01,620 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loading jobs for terms
Exception in thread "ConsistencyCheckerRunner" java.lang.IllegalArgumentException: $in operator is not allowed. Allowed options are: [$gte, $eq, $neq, $lt, <=, !=, $lte, >=, >, =, <]
        at com.redhat.lightblue.client.expression.query.ValueQuery.<init>(ValueQuery.java:37)
        at com.redhat.lightblue.client.expression.query.ValueQuery.withValue(ValueQuery.java:75)
        at com.redhat.lightblue.migrator.consistency.ConsistencyChecker.getMigrationJobs(ConsistencyChecker.java:219)
        at com.redhat.lightblue.migrator.consistency.ConsistencyChecker.run(ConsistencyChecker.java:145)
        at java.lang.Thread.run(Thread.java:745)
2015-03-27 15:42:04,974 [WorkerMonitor] com.redhat.lightblue.migrator.consistency.ConsistencyCheckerDaemon WARN - ConsistencyChecker has stopped running, attempting restart.

```

### PR DESCRIPTION
**Pull request by dcrissman**
Creates an integration test framework that we can build on for writing migrator integration tests.

This PR really only adds the necessary bones to start writing integration tests. Future PRs will contain additional tests.

Depends on https://github.com/lightblue-platform/lightblue-test-utils/pull/16

### CODE REVIEW

File: consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private static boolean initialized = false;
+
+    private static String versionMigrationJob;
+    private static String versionMigrationConfiguration;
+
+    public ITCaseConsistencyCheckerTest() throws Exception {
+        super();
+
+        if (!initialized) {
+            loadData("migrationConfiguration", versionMigrationConfiguration, "./test/data/load-migration-configurations.json");
```
Comment (jewzaam) [inline consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java]: This isn't thread safe.  Problem if tests are run in multiple threads?

File: consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        consistencyChecker.setMigrationJobEntityVersion(versionMigrationJob);
+    }
+
+    @Override
+    protected JsonNode[] getMetadataJsonNodes() throws Exception {
+        ObjectNode jsonMigrationJob = (ObjectNode) loadJsonNode("./migrationJob.json");
+        ObjectNode jsonMigrationConfiguration = (ObjectNode) loadJsonNode("./migrationConfiguration.json");
+
+        versionMigrationJob = parseEntityVersion(jsonMigrationJob);
+        versionMigrationConfiguration = parseEntityVersion(jsonMigrationConfiguration);
```
Comment (jewzaam) [inline consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java]: How does this work?  The version static vars are set after they're needed in the constructor.

Comment (jewzaam) [inline consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java]: Oh, that isn't a test method.  Is there something in super() that is triggering this to be called?

Comment (dcrissman) [inline consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java]: Yes, it is up the chain all the way to the top. Simply, this method will be called by the super() constructor.

File: consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java (hunk 1 above)
Comment (dcrissman) [inline consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java]: I don't think junit executes in multiple threads... does it? If this is a problem, then some other stuff further up the chain will also have issues.

Comment (jewzaam) [inline consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/ITCaseConsistencyCheckerTest.java]: You can setup junit to fork, which I think does something with multithreading.  Maybe not per class though.

**Pull request merged**